### ISSUE TITLE
lost feed

### ISSUE THREAD

**Issue by plauters** [ancestorak/brief#138]
Good evening,

I'm looking for some help with Brief. Yesterday, all my items in Brief disappeared, the bookmarks too, when I tried to add a new feed to it.
I tried to restore them but had no luck at the moment. I "just" wish to find my old items back, or a way to save them in archives.
I tried to find some help on various websites but couldn't see any solution at the moment

Brief version 1.6.2

Any help appreciated,
Thanks

**Comment by tanriol** [comment]
What is your Firefox version? Brief 1.6.2 is *severely* outdated and not expected to work correctly now, please try updating to [Brief 2.0b3](https://addons.mozilla.org/firefox/downloads/file/294808/brief-2.0b3-fx.xpi?src=dp-btn-devchannel).

**Comment by plauters** [comment]
Firefox 35.0.1
I never realised my version of Brief was so outdated, I expect to be notified when new versions are available. Well.
I installed Brief 2.0b3: it downloaded the new feeds only. The old items and favorites are still missing.

**Comment by tanriol** [comment]
Unless you have switched to the view unreads only mode (have you?), it sounds like `brief.sqlite` got corrupt or missing... unfortunately, there is no universal method of fixing this. You may try going to the Firefox profile folder and copying `brief-backup.sqlite` over `brief.sqlite`. If you do, don't forget to backup them!

**Comment by AussieD** [comment]
@plauters, it's also worth noting that the original developer of Brief (ancestorak) doesn't appear to be very active in maintaining it any more (or at least inactive enough to be away for extended periods). As a result, tanriol appears to have kindly picked up the slack. But since he is not the original author, current updates are not being uploaded to the Mozilla Add-ons store, unless ancestorak does it.

So, to stay current (for the forseeable future) you'll need to manually perform the updates as they are pre-released here on GitHub. If you add https://github.com/ancestorak/brief/releases as an RSS feed you'll be in the loop when when new pre-release versions are made available.

Don't forget to make sure you have not accidentally filtered to only show unread news items.

**Comment by plauters** [comment]
Nope, I'm sure I chose to see the "read items" too. They were simply not there.
I made a copy of a brief-backup.sqlite over brief.sqlite and it restored some of them. Unfortunately that backup was 4 years old, so I have lost all the items between the backup and the new ones.

Thank you very much for your help.

**Comment by Marooned-MB** [comment]
After switching from 1.6.2 to Brief 2.0b3 I've lost some old fav items but feed was still working.. for a few days. A week ago or so all my feeds disappeared. Not only items, all feeds on the left. Like I don't have any subscriptions. I've noticed that option panel is now very simple, no Feed tab.

I've also noticed new "Subscribed Feeds" folder in bookmarks that is empty and is recreated when deleted. Does it means that new Brief has this folder hardcoded and we can't use our own like it was till now?

**Comment by tanriol** [comment]
Yes, @ancestorak removed the UI for choosing the home folder. Technically it can still be changed (the Brief's `homeFolder` pref stores the ID of that folder), but the UI is gone.

If you see the "Subscribed Feeds" folder and it is recreated, Brief is using it as the home folder. Actually, you can rename or move it as you wish, and Brief will work with it despite the changes. Well, at least if it does not, then this is a bug and will have to be fixed. Thus almost everything is still possible, except probably forcing Brief on multiple devices to use the same folder... which would be quite useless as Brief doesn't sync either the items or their unread state yet.

**Comment by Marooned-MB** [comment]
So that change seems illogical as it just makes all feeds disappear. Would have more sense to copy all subscription there so user won't notice but Brief should be just a viewer. It should not force user to use folder picked by Brief's author nor to access config. I can do this, you can do this, but there are many people using computer as black box and it's black magic for them. It's like "it was working, now it's not".

**Comment by tanriol** [comment]
Brief is not intended to change the folder if it already exists. It is a bug.

**Comment by tanriol** [comment]
> Hmm, `extensions.brief.homeFolder` is integer in my config, not a path.

Yes, the folder ID is an internal Firefox identifier, which does not change due to renames and moves. It is really not for manual configuration, just for persistent storage. The easiest workaround seems to be to move all the feeds to the Subscribed Feeds folder and then move and/or rename it as you need.

**Comment by Marooned-MB** [comment]
Ok, thanks for the help, but with all those glitches (I really liked Brief and used it for few years) I'm just testing alternative feed extension. Don't want to loose all my favs again as it already happen.
Thanks anyway.

**Comment by sleek22** [comment]
For those of you using the latest beta version 2.0b3 and have no way of pointing to your "Subscribed Feeds" folder, here is a solution:

( In my case, they disappeared when I had done a bookmarks restore. This means the folder ID numbers had changed )


Install Add Bookmark Here2 addon. Configure it to use your "Subscribed Feeds" folder as its default folder. (Advanced Option - Customize the Unsorted Bookmarks Folder) - There is a drop-down menu with your bookmarks folder shown.

Do this with about:config open in the background and have a search opened on "folderID".

You will see the following key: extensions.abhere2.folderId.unsorted

You will see it's value change instantly as you click on each folder in the configuration menu. Mark down the folder ID number when you have chosen your "Subscribed Feeds" folder.

Now do another search in about:config on extensions.brief.homeFolder

Place the number you obtained from doing the previous step as the folder ID value for extensions.brief.homeFolder

Viola! Your folders are back.


https://addons.mozilla.org/en-US/firefox/addon/add-bookmark-here-2/

**Comment by tanriol** [comment]
> In my case, they disappeared when I had done a bookmarks restore. This means the folder ID numbers had changed

Oops! Thank you very much, I had never thought about bookmark IDs changing during backup and restore operation.

**Comment by sleek22** [comment]
Here is another method for finding the folderID you need to put in about:config for extensions.brief.homeFolder. This is easier and uses SQLite Manager.

In SQLite Manager, open the places.sqlite database. Click on moz_bookmarks. Do a search for "Subscribed Feeds" or the name of the folder you have your feeds in. You will find the folder ID there.

https://addons.mozilla.org/en-US/firefox/addon/sqlite-manager/

**Comment by Marooned-MB** [comment]
Now that you mentioned it i must admit I also did bookmarks restore.

**Comment by NETHeader** [comment]
I've also ran into that problem today, and I've had to search quite a while why brief is malfunctioning (no info at the homepage, no error message, etc).
Couldn't it be solved by simply bringing the UI for choosing the home folder back?

**Comment by sleek22** [comment]
@NETHeader,

The simplest solution is to have "Subscribed Feeds" be the default folder for Brief. It's a good as name as any. Have Brief search for this folder on either browser start-up or when Brief is opened. Create the folder if not found. That way, Brief would be attaching to a folder name and not a folder number, which as noted above, changes when bookmarks are restored.

**Comment by NETHeader** [comment]
@sleek22
Now that I know where to look for, this is not a problem anymore to me, since you have already posted a way to solve this. I've already changed "extensions.brief.homeFolder" to the correct folder id, and it has worked as expected right out of the box.

The folder "Subscribed Feeds" is called "Abonnierte Feeds", since my default language is german.
But nevertheless, this folder was present all the time - I think the underlying problem seems that firefox replaces the folder (the folder id) without any chance for brief to reflect this change.

My problem had been introduced as I changed my OS from Windows to Debian. As far as i remember, I've already installed Brief at the new system and then imported my bookmarks (of course with the default folder "Subscribed Feeds" or "Abonnierte Feeds") from my old firefox installation -- so I guess it is not inexplainable why brief ran into this problem. Firefox may have replaced the id, leaving the folder name the same as before.

I think it's not an uncommon method to backup/restore the settings this way, so I'm expecting also other users to be affected -- some of the negative ratings at [https://addons.mozilla.org] indicate this already.


> Have Brief search for this folder on either browser start-up or when Brief is opened.

Sorry to say this, but in my eyes this is no solution, it is a poor (because short-dated) workaround.
If one would implement this, on the one side, Brief have to consider a lot of different languages, and on the other side this does not solve the problem, since the user is free to change the name of the default folder anytime he/she wants.
There is a good reason why firefox and brief using only the folder id, and don't rely on the folder name itself.

This seems a special, but systematical problem within Brief, isn't it?

I can not fully understand what the decisions behind a removal of the choosing GUI were (why not simply leaving this as an configuration option?), but I hope this can be solved without introducing a lot of side effects and/or additional code.
Or---if that is not an option---just mentioning the problem (and your workaround solution) a little bit more prominent at the [Addon-Homepage](http://brief.mozdev.org/).
One have to dig really deep into the forum or here at github to a) identify the problem and b) to solve it.

**Comment by sleek22** [comment]
@NETHeader ,

I didn't think about the language issue. I guess we're back to an option in the UI allowing us to choose the home folder as before.

**Comment by bobular** [comment]
I had to follow NETHeader's advice (thanks! it worked) to fix extensions.brief.homeFolder in about:config - the reason being that I use firefox sync between different browsers and I really had to point brief back to my sync'ed "Subscribed feeds" because Brief was using the new empty "Subscribed feeds" on my "new" browser.  It would be really great to configure this via UI again!  Thanks

**Comment by tanriol** [comment]
I tend to agree that this option makes sense. However, the previous UI for it was... not exactly nice. Any suggestions on a nice UI solution, small yet discoverable? If possible, integrated into the Firefox stock Library view... can we make it easy to discover if needed?

**Comment by silvio** [comment]
I still have this problem. Can we have the UI back for selecting what the folder for brief is? I have 5 firefox installations (mobile, work laptop, work vm, home desktop and home laptop) and at least 5 "subscribed feeds" folder.

**Comment by tanriol** [comment]
Sorry, I'm a bit overloaded right now. Will certainly consider this request.

**Comment by silvio** [comment]
Is eb61a401b256b43024268f38ede681d495e93930 the commit which remove selectable of the feed-folder?

**Comment by silvio** [comment]
see #156

**Comment by Bundaboy** [comment]
I am finding the ID of the Subscribed Feeds Folder has suddenly started changing of it's own accord (no backup/restore or sync) usually when FF restarts but sometimes in the middle of a session. Has anyone else had this problem? The Subscribed Feeds Folder gets recreated but, of course, I have to then copy my feeds across to it.
